db/wal_manager.cc

```diff
@@ -324,6 +324,13 @@ Status WalManager::GetSortedWalsOfType(const std::string& path,
         }
       }
       if (!s.ok()) {
+        if ((s.IsNotFound() || s.IsIOError()) && log_type == kArchivedLogFile) {
```

Comment:
I/O error could also mean something else. Maybe in case of I/O error we should verify `env_->FileExists(path).IsNotFound()` before suppressing the error and continuing?